### DISCUSSION TITLE
Does Supabase Support macOS/Desktop Applications in Flutter?

### DISCUSSION THREAD

**Discussion by Asmaa-alt** [supabase/supabase#26714] [Questions] [answered]
Hello Supabase Community,

I am currently developing a macOS desktop application using Flutter and attempting to integrate Supabase for authentication and database operations. While my requests work perfectly on the web version of my application, they fail to function on the desktop version.

Specifically, I have implemented user sign-up and data fetching functionalities. These operations perform as expected when tested on the website, but when I run the same code on the macOS desktop application, the requests do not seem to execute properly and I am  I'm consistently receiving an AuthRetryableFetchError.

Has anyone else encountered similar issues? Is there something specific I need to configure or consider when using Supabase with Flutter on macOS?

Any guidance or suggestions would be greatly appreciated!

Thank you!

**Comment by dshukertjr** [comment, marked as answer]
The error has nothing to do with Supabase. In order to make HTTP requests from desktop applications, there is a configuration that you need to turn on to allow it. You can turn it on, and Supabase will work just fine.

**Comment by akagaeng** [comment]
@Asmaa-alt 

```
# add code in below
# to macos/Runner/DebugProfile.entitlements and macos/Runner/Release.entitlement

<key>com.apple.security.network.client</key>
<true/>
```

Alternatively, open `macos/` from your XCode and select `Outgoing Connections (Client)`.

**Reference**
* https://stackoverflow.com/questions/57841871/flutter-http-get-fails-on-macos-build-target-connection-failed

It worked for me.
